### DISCUSSION TITLE
Theming and Font Issues for Qt Flatpak and Snaps

### DISCUSSION THREAD

**Discussion by ghost** [lxqt/lxqt#2376] [General]
I am using latest LXQT 1.2.0 in Lubuntu 22.04.1, LXQT has a lot of themeing issues, like GTk applications and even Qt apps like pcmanfm-qt when run as root doesnt respect user theme, these is also visible in Flatpak apps, Flatpak Qt and Gtk apps doesnt use my configured font sizes.
So to summarise i need help and attention toward the following -
1) Proper LXQT themeing for Native Root Gtk and Qt apps(Native means deb apps from repos)
2) Proper LXQT font and theme configuration for Qt and Gtk apps of Sandboxed nature likde Flatpaks.
3) Some LXQT tabs have broken and shoddy titles and descriptions, also LXQT notification improperly draws the cross/cut sign used to dismiss the Notifications(Images Attached).

##### Expected Behavior
I expect LXQT to properly theme apps both Qt and Gtk while using root session, i want LXQT to provide uniform and proper font config for flatpaks and want to report about broken LXQT Titles and Indicators/Icons.

##### Current Behavior
Qt and Gtk apps Disrespect User configured Theming when using root sessions.
Flatpak apps of any nature(Gtk,Qt,Libadwaita) wont respect my Font Config
LXQT components seem to be broken.

##### Steps to Reproduce (for bugs)
1. Use Root session of featherpad,pacmanfm-qt,Timeshift,Gdebi install session etc., i have configured Papirus-Dark and Kvantum Materia Dark for Qt and Yaru-Dark  for Gtk but these apps doesnt use Dark theme and uses old and ugly icons instead of Papirus-Dark.
2. Flatpak apps like Okular,SMPlayer,Gnome text Editor,Clapper etc. wont use my Font Config(Ubuntu Regular 13)
3. LXQT DEsktop preferences has a Slide Show tab where the titles appear Broken, needs fixing.
4. Notifications appearing have broken cut/cross sign/icon.

![Bug1](https://user-images.githubusercontent.com/100503303/215480612-a83b3cef-0b5c-40fb-a76a-087bb94e0b60.jpg)

![Bug2](https://user-images.githubusercontent.com/100503303/215481362-e3a38eab-8989-4e72-aa2a-59d1f747359d.jpg)

![Bug3](https://user-images.githubusercontent.com/100503303/215481413-e4ad9236-1a4a-4851-ba42-46e6663244cf.jpg)

![Bug 3](https://user-images.githubusercontent.com/100503303/215481461-f32acc99-b158-42fb-bda5-fca48904be4b.jpg)

##### Context
 Trying to properly theme apps and have a proper Font Config for all types of apps across the os as other Desktop Environments have.

##### System Information

* Distribution & Version: Lubuntu 22.04.1 with latest Lxqt from Lubuntu Backports Repo
* Kernel:  5.15
* Qt Version: 5.15.3 as per about Lxqt
* liblxqt Version: 
* Package version: 1.2.0

Info from About Lxqt--
LXQt Desktop Toolbox - Technical Info<p>
LXQt About Version:  1.2.0
LXQt Version:  1.2.0
Qt:  5.15.3
Build type:  Release
System Configuration: /etc/xdg
Share Directory: /usr/share/lxqt
Translations:  /usr/share/Lubuntu/lxqt/translations 
/usr/share/gnome/lxqt/translations
/home/sam/.local/share/flatpak/exports/share/lxqt/translations
/var/lib/flatpak/exports/share/lxqt/translations
/usr/local/share/lxqt/translations
/usr/share/lxqt/translations
/var/lib/snapd/desktop/lxqt/translations

User Directories
Xdg Data Home:          /home/sam/.local/share
Xdg Config Home:        /home/sam/.config
Xdg Data Dirs:          /usr/share/Lubuntu
/usr/share/gnome
/home/sam/.local/share/flatpak/exports/share
/var/lib/flatpak/exports/share
/usr/local/share
/usr/share
/var/lib/snapd/desktop
Xdg Cache Home:         /home/sam/.cache
Xdg Runtime Home:       /run/user/1000
Xdg Autostart Dirs:     /etc/xdg/xdg-Lubuntu/autostart
/etc/xdg/autostart
/etc/autostart
/usr/share/autostart
Xdg Autostart Home:     /home/sam/.config/autostart

**Comment by stefonarch** [comment]
Any  LXQt application opened as root uses  `/root/.config/lxqt` which usually is not configured as there is no reason to run any graphic app as root. You can use featherpad to open a file in /etc and modify it, hit "save" and it will ask you for authorization, pcmanfm-qt has "open tab as root" and "open tab as admin" and so on.

You can copy` ~/.config/lxqt `to` /root/.config/ `to have similar look, same for gtk settings and so on, but as I said IMHO there is no reason for using those apps as root.

I convert that into a discussion, the flatpak issue could be more investigated, I've none of them.

**Comment by tsujan** [comment]
To style root apps, you need to use root settings. Root is another user. Linux/Unix is a multi-user system, and each user has its own settings.

As for Flatpak, the problems should be reported to Flatpak.

> LXQT DEsktop preferences has a Slide Show tab where the titles appear Broken, needs fixing.

What you see should be a problem in the Qt theme you use.

**Reply by stefonarch** [reply]
I've tried flatpak's strawberry in VM with Lubuntu and in fact it looks out of place, including mouse cursor. For Qt theming it looks like something like this is needed: 
https://www.reddit.com/r/linux/comments/psit31/flatpak_now_supports_qt_theming_with_kvantum/

**Comment by ghost** [comment]
The Flatpak one is a issue exclusive to Lxqt desktops, i again experienced this with Vlc Snap Also, i think the problem is that Lxqt lacks or is unable to specify system configuration of fonts to snadboxed apps as for the title i tried it with different themes yet no result upon closely looking at it it seems its crashing with some other title like lubuntu help or something

**Reply by tsujan** [reply]
> i think the problem is that Lxqt lacks or mis unable to specify system configuration of fonts to snadboxed apps

Flatpak apps should behave like other apps and follow the user settings. If not, they have problems.

**Reply by ghost** [reply]
But IMHO you can at least suggest a way how does other Desktop Environments do it ?? I mean it will overall help the DE itself as the Linux Community is aggressively moving towards Sand-boxed Apps.....

**Reply by tsujan** [reply]
Packaging is a downstream job. In no way LXQt is responsible for it.

The sooner such issues are reported to Flatpak, the sooner they might be fixed.

**Reply by ghost** [reply]
But they will send me back like you did that report it downstream since its not an issue with Flatpak, it depends on System config which other DEs have successfully implemented, i mean i dont have a reason to report to them this when my own DE doesnt know how to implement something which is not problematic, nevertheless i will try at Lubuntu Forums..

**Reply by tsujan** [reply]
> But they will send me back like you did that report it downstream

Quote my previous comment. They shouldn't except downstream jobs to be done by upstream developers; if they do (I'm not sure because I don't know them), any discussion will be useless.

**Reply by ghost** [reply]
Ok let me see, anyways can you help out with Gtk apps font appearing dull (Native apps) and in this Flatpak Smughead i forgot my other main issue that is gtk applications needing root access like Timeshift etc. they don't respect theme set by me and i did check root config but all files of lxqt and gtk are already there.

**Reply by tsujan** [reply]
What I know for sure is that GTK3's root settings are in `/root/.config/gtk-3.0/` and that they work here. Anything beyond that should be done by `dconf-editor`, but I don't know whether it can/should be run as root or how.

I don't know about GTK4. Sorry, I'm not an expert in GTK — hopefully, another member/user will answer your GTK questions.

**Comment by tsujan** [comment]
BTW, if your screenshot of Desktop Preferences was taken with a Kvantum theme, please attach the theme as a zipped file for me to check it. It isn't MateriaDark.

**Reply by ghost** [reply]
Oh yes sure its called KvLibadwaita, its not a zip file i cloned it and installed, here is the link : https://github.com/GabePoel/KvLibadwaita

**Reply by tsujan** [reply]
Thanks! The theme seems OK. What I get here is very different from [your screenshot](https://user-images.githubusercontent.com/100503303/215480612-a83b3cef-0b5c-40fb-a76a-087bb94e0b60.jpg):

![desktop_pref](https://user-images.githubusercontent.com/981076/215523945-751b42ca-ac76-4c3d-be41-47057932bd32.png)

Probably you'd downloaded an old version of it.

**Reply by ghost** [reply]
Git cloning downlaods the latest version right the problem here i think is something  was extra added by the Lubuntu Community...

**Reply by tsujan** [reply]
Maybe. I was just curious about how a Kvantum theme might result in your screenshot, because, theoretically, it shouldn't.

**Comment by ghost** [comment]
Hey there community, i need some help regarding sym-linking /home/xyz/.config/(folders like gtk3.0,lxqt,kvantum) with /root/.config, so that updating the home one automatically changes the root one, i only need to connect some specific folders to improve theming consistency for  root apps like timeshift,Bleachbit, i have manually added all the needed folders from /home/xyz/.config to .config/2root but i need some settings which automatically syncs the folders and file between them.

**Reply by stefonarch** [reply]
I think you need a script and you need to run it as root, not sure if a symlink in `/root/.config/foo ` to `~/.config/foo` will work, but you could try it.

**Comment by ghost** [comment]
Added a page in lxqt wiki on Gtk theming in Lxqt -: https://github.com/lxqt/lxqt/wiki/Configuring-Proper-Gtk-theming-in-Lxqt-for-Icons,Fonts-and-Theme

**Reply by stefonarch** [reply]
Thanks for looking in container theming issues and the wiki page, which has several issues IMO.

* extensive formatting, "talk" language. Wiki pages should be consistent among them
* no need for `sudo` to create a  file in `$HOME`, no need for passwd 

For theme changes  maybe we need `lxqt-config-appearance` writing also  to `~/.config/xsettingsd/xsettingsd.conf` ? Is configuring `xsettingsd` something that LXQt should do? My guess is that this should be done by distros.

**Reply by ghost** [reply]
My intention was to keep the wiki page newbie friendly so i used non-technical language but yes consistency should have been maintained, sorry for the sudo, since i thought it to be a system level config file and remaining for configuring xsettingsd i think lxqt must do it since in case any distro forgets to implement it users will only consider it a bug or lack in lxqt  not in the distro itself and yes it will also overall improve the way gtk/libadwaita apps exist in Lxqt

**Reply by ghost** [reply]
Can you also please take out some time to help me figuring out something like xsettingsd for Qt apps that i can expose to flatpak apps so that they respect my font config, theming Flatpak Qt apps can be achieved by giving access to Kvantum but the fonts look ugly...

**Reply by stefonarch** [reply]
I looked into flatpak some day ago and after filling up ` /var/lib/ `with 6gb of stuff for 3 apps I removed all of it.  Font in qt flatpaks seems an issue not easy to solve; mouse cursor was ok after giving permissions to read `$HOME' in flatseal.

If Redhat push their things it's up to them to fix issues IMO.

**Reply by tsujan** [reply]
Flatpak is for sandboxing, i.e., running apps in isolation from the rest of the system. So, these issues are expected. "Solving" them contradicts the purpose of Flatpak.

**Reply by tsujan** [reply]
> ... such futuristic technology

That's the misunderstanding. It's as true as an advertisement can be.

**Reply by tsujan** [reply]
> Just imagine a situation....

I might need to imagine these things if I was a newbie.

> Isn't this Futuristic

No, it's rather regressive.

**Comment by tsujan** [comment]
Since Flatpak is in the title of this discussion, let's be more explicit.

You may use Flatpak only to run apps in isolation. Then, you couldn't expect them to follow the settings of your system — Flatpak itself doesn't make such a promise (actually, it makes the opposite promise). And Flatpak doesn't see it like a bug when your Flatpak apps don't use your system theme/font/icons….

The same is true for other sandboxing methods.

You may insist on using Flatpak and add some workarounds for Flatpak apps to behave like ordinary apps *as far as possible*. But there will always be a difference.

**Reply by tsujan** [reply]
> which has been successfully achieved for gtk flatpaks in gtk based DEs and Qt flatpaks in Kde

I don't have to say anything about GTK-based DE's — they even don't have proper integration for Qt apps.

As for KDE, see https://www.reddit.com/r/kde/comments/ujqih9/plasma_consistent_flatpak_theming/. If Flatpak needs such maneuvers for an elementary thing like styling and yet the result has basic problems, then either it's a bad joke or it isn't supposed to do that.

**Comment by AzumaHazuki** [comment]
I am old enough to remember when this concept didn't need an FS overlay and was called "static linking" =P

Kidding, kidding! But really, this seems like a bad way of reinventing the wheel. If nothing else, why isn't there a sort of "portal" for Flatseal that, say, reads one's xsettings.conf file?
